url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/Statement.java

```diff
@@ -0,0 +1,610 @@
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+//
+
+package org.msgpack.template.builder.beans;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import org.apache.harmony.beans.BeansUtils;
+import org.apache.harmony.beans.internal.nls.Messages;
+
+public class Statement {
+
+    private Object target;
+
+    private String methodName;
+
+    private Object[] arguments;
+
+    // cache used methods of specified target class to accelerate method search
+    private static WeakHashMap<Class<?>, Method[]> classMethodsCache = new WeakHashMap<Class<?>, Method[]>();
+
+    public Statement(Object target, String methodName, Object[] arguments) {
+        this.target = target;
+        this.methodName = methodName;
+        this.arguments = arguments == null ? BeansUtils.EMPTY_OBJECT_ARRAY
+                : arguments;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        if (target == null) {
+            sb.append(BeansUtils.NULL);
+        } else {
+            Class<?> clazz = target.getClass();
+            sb.append(clazz == String.class ? BeansUtils.QUOTE : BeansUtils
+                    .idOfClass(clazz));
+        }
+        sb.append('.' + methodName + '(');
+        if (arguments != null) {
+            Class<?> clazz;
+            for (int index = 0; index < arguments.length; index++) {
+                if (index > 0) {
+                    sb.append(", "); //$NON-NLS-1$
+                }
+                if (arguments[index] == null) {
+                    sb.append(BeansUtils.NULL);
+                } else {
+                    clazz = arguments[index].getClass();
+                    sb.append(clazz == String.class ? '"' + (String) arguments[index] + '"'
+                            : BeansUtils.idOfClass(clazz));
+                }
+            }
+        }
+        sb.append(')');
+        sb.append(';');
+        return sb.toString();
+    }
+
+    public String getMethodName() {
+        return methodName;
+    }
+
+    public Object[] getArguments() {
+        return arguments;
+    }
+
+    public Object getTarget() {
+        return target;
+    }
+
+    public void execute() throws Exception {
+        invokeMethod();
+    }
+
+    Object invokeMethod() throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 28. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.